src/main/java/org/opentripplanner/transit/model/timetable/TripTimes.java

```diff
@@ -1,618 +1,172 @@
-
 import java.io.Serializable;
 import java.time.Duration;
-import java.util.Arrays;
+import java.util.Comparator;
 import java.util.List;
-import java.util.Optional;
 import java.util.OptionalInt;
-import java.util.function.IntUnaryOperator;
 import javax.annotation.Nullable;
 import org.opentripplanner.framework.i18n.I18NString;
 import org.opentripplanner.model.BookingInfo;
 import org.opentripplanner.transit.model.basic.Accessibility;
 
-/**
- * A TripTimes represents the arrival and departure times for a single trip in an Timetable. It is
- * carried along by States when routing to ensure that they have a consistent, fast view of the trip
- * when realtime updates have been applied. All times are expressed as seconds since midnight (as in
- * GTFS).
- */
-public final class TripTimes implements Serializable, Comparable<TripTimes> {
-
-  private ScheduledTripTimes scheduledTripTimes;
-
-  private int[] arrivalTimes;
-  private int[] departureTimes;
-  private RealTimeState realTimeState;
-  private StopRealTimeState[] stopRealTimeStates;
-  private I18NString[] headsigns;
-  private OccupancyStatus[] occupancyStatus;
-  private Accessibility wheelchairAccessibility;
-
-  private TripTimes(final TripTimes original, int timeShiftDelta) {
-    this(
-      original,
-      original.scheduledTripTimes.copyOfNoDuplication().plusTimeShift(timeShiftDelta).build()
-    );
-  }
-
-  TripTimes(ScheduledTripTimes scheduledTripTimes) {
-    this(
-      scheduledTripTimes,
-      scheduledTripTimes.getRealTimeState(),
-      null,
-      null,
-      null,
-      scheduledTripTimes.getWheelchairAccessibility()
-    );
-  }
-
-  private TripTimes(TripTimes original, ScheduledTripTimes scheduledTripTimes) {
-    this(
-      scheduledTripTimes,
-      original.realTimeState,
-      original.stopRealTimeStates,
-      original.headsigns,
-      original.occupancyStatus,
-      original.wheelchairAccessibility
-    );
-  }
-
-  private TripTimes(
-    ScheduledTripTimes scheduledTripTimes,
-    RealTimeState realTimeState,
-    StopRealTimeState[] stopRealTimeStates,
-    I18NString[] headsigns,
-    OccupancyStatus[] occupancyStatus,
-    Accessibility wheelchairAccessibility
-  ) {
-    this.scheduledTripTimes = scheduledTripTimes;
-    this.realTimeState = realTimeState;
-    this.stopRealTimeStates = stopRealTimeStates;
-    this.headsigns = headsigns;
-    this.occupancyStatus = occupancyStatus;
-    this.wheelchairAccessibility = wheelchairAccessibility;
-
-    // We set these to null to indicate that this is a non-updated/scheduled TripTimes.
-    // We cannot point to the scheduled times because we do not want to make an unnecessary copy.
-    this.arrivalTimes = null;
-    this.departureTimes = null;
-  }
-
-  public static TripTimes of(ScheduledTripTimes scheduledTripTimes) {
-    return new TripTimes(scheduledTripTimes);
-  }
-
+public interface TripTimes extends Serializable, Comparable<TripTimes> {
```

Comment:
Can you provide a quick description what this interface abstracts over?

Reply:
It seems The Javadoc currently in RealTimeTripTimes should be moved to TripTimes

Reply:
7d052cb